### PR TITLE
Graph skeleton

### PR DESCRIPTION
**Pull request by kathtran**
Added two modules (graph and projectquire) and respective test files for each.

### CODE REVIEW

File: lib/graph.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+'use strict';
+let projectquire = require('./projectquire');
+let Ajv = require('ajv');
+
+let schema = require('./relationshipGraphSpec');
```
Comment (ibilal) [inline lib/graph.js]: This will be under the schema folder in lib

File: lib/graph.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+class Graph {
+  /**
+   * Builds a graph object model given a project ID and a url that contains the username, password, and team name.
+   *
+   * @param {Number} projectId
+   * @param {String} url Should be in the following format: http://username:password@teamname.jamacloud.com/rest/latest/
+   * @constructor {Graph}
+   */
+  constructor (projectId, url) {
+    this.name = '';
```
Comment (ibilal) [inline lib/graph.js]: Is this the project's name? I'd name it as such like the param projectId

File: lib/graph.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   * Returns an object model abiding by the relationship graph JSON schema from the given nodes and edges.
+   *
+   * @returns {{items: Array, relationships: Array}} if nodes and edges is in the proper format, undefined otherwise
+   */
+  toJson () {
+    let graphJson = { 'items': this.nodes, 'relationships': this.edges };
+    let ajv = new Ajv();
+    var valid = ajv.validate(schema, graphJson);
+    if (!valid) {
+      return { 'items': this.nodes, 'relationships': this.edges };
```
Comment (ibilal) [inline lib/graph.js]: Are you returning the graph nodes and edges when valid is false?

Comment (ibilal) [inline lib/graph.js]: Are you returning the graph nodes and edges when valid is false?

File: lib/pagination.js [hunk 4]
```diff
@@ -7,28 +7,40 @@ function pagination (baseUrl, startAt, maxResults) {
...
+                if (payload.length < max) {
+                  payload.push(item);
+                }
+              });
+              let pageInfo = response.body.meta.pageInfo;
+              start = pageInfo.startIndex + PAGE_SIZE;
+              resultCount = (payload.length <= max) ? pageInfo.resultCount : 0;
+              return getPages(baseUrl, start, max);
+            } else {
+              payload.push(response.body.data);
```
Comment (ibilal) [inline lib/pagination.js]: what is response.body.data if its not an instance of array?

File: lib/graph.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    } else {
+      console.error('Graph is not returning a valid graph relationship object');
```
Comment (ibilal) [inline lib/graph.js]: Your tests expect an empty json, is returning nothing equivalent?

File: test/lib/projectquire.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  let rejectedPromise = () => {
+    return new Promise((resolve, reject) => {
+      reject(data);
+    });
+  };
+
+  describe('getProjectName function', () => {
+    it('should return a resolved promise when there exists a project name.', () => {
+      projectquire = proxyquire('../../lib/projectquire', {'./pagination': resolvedPromise});
+      expect(projectquire.getProjectName(projectId, url)).to.be.fulfilled();
```
Comment (ibilal) [inline test/lib/projectquire.js]: what is data set to in this test?

File: lib/graph.js (hunk 2 above)
Comment (RickyV33) [inline lib/graph.js]: this gets named based on the projectId, happens at line 20

File: lib/pagination.js (hunk 4 above)
Comment (RickyV33) [inline lib/pagination.js]: If a single item is returned from jama, such as when you request a project name based on the id, it is not stored in an array object. This handles that

**Comment by RickyV33** [comment]
travis-ci will fail since it does not include the schema for the relationship graph

**Pull request merged**